pkg/tokens/builtin/zoneegress/token.go

```diff
@@ -0,0 +1,34 @@
+package zoneegress
+
+import (
+	"github.com/golang-jwt/jwt/v4"
+
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+	core_tokens "github.com/kumahq/kuma/pkg/core/tokens"
+)
+
+const ZoneEgressSigningKeyPrefix = "zone-egress-token-signing-key"
+
+var ZoneEgressTokenRevocationsGlobalSecretKey = core_model.ResourceKey{
+	Name: "zone-egress-token-revocations",
+	Mesh: core_model.NoMesh,
+}
+
+type zoneEgressClaims struct {
+	Zone string
+	jwt.RegisteredClaims
+}
+
+var _ core_tokens.Claims = &zoneEgressClaims{}
+
+func (c *zoneEgressClaims) ID() string {
+	return c.RegisteredClaims.ID
+}
+
+func (c *zoneEgressClaims) KeyIDFallback() (int, error) {
+	return 0, nil
```

Comment:
return error here. We don't want to support lack of `kid` for new tokens

Reply:
addressed